irohad/ametsuchi/impl/mutable_storage_impl.cpp

```diff
@@ -23,48 +22,75 @@ namespace iroha {
         std::shared_ptr<shared_model::interface::CommonObjectsFactory> factory)
         : top_hash_(top_hash),
           sql_(std::move(sql)),
-          wsv_(std::make_shared<PostgresWsvQuery>(*sql_, factory)),
+          peer_query_(std::make_unique<PeerQueryWsv>(
+              std::make_shared<PostgresWsvQuery>(*sql_, factory))),
           block_index_(std::make_unique<PostgresBlockIndex>(*sql_)),
           command_executor_(std::make_shared<PostgresCommandExecutor>(*sql_)),
           committed(false),
           log_(logger::log("MutableStorage")) {
       *sql_ << "BEGIN";
     }
 
-    bool MutableStorageImpl::check(
-        const shared_model::interface::Block &block,
-        MutableStoragePredicate predicate) {
-      PeerQueryWsv peer_query(wsv_);
-      return predicate(block, peer_query, top_hash_);
-    }
-
-    bool MutableStorageImpl::apply(const shared_model::interface::Block &block) {
+    bool MutableStorageImpl::apply(const shared_model::interface::Block &block,
+                                   MutableStoragePredicate function) {
       auto execute_transaction = [this](auto &transaction) {
         command_executor_->setCreatorAccountId(transaction.creatorAccountId());
         command_executor_->doValidation(false);
+
         auto execute_command = [this](auto &command) {
           auto result = boost::apply_visitor(*command_executor_, command.get());
+
```

Comment:
Strange empty line :(

Reply:
Why? It looks more readable to me, since it separates function return statement.